### PR TITLE
Allow the download of cases even if some selected rows are not cases.

### PR DESCRIPTION
**Pull request by EstherDarkish**

### CODE REVIEW

File: src/components/utils/caseUtils.ts [hunk 1]
```diff
@@ -17,3 +27,136 @@ export const downloadCases = async (uuids: string[]) => {
...
+        if (numberOfDownloadedCases === 1) {
+            messageFirstLine = intl.formatMessage({
+                id: 'download.message.downloadedCase',
+            });
+        } else {
+            messageFirstLine = intl.formatMessage(
+                { id: 'download.message.downloadedCases' },
+                {
+                    number: numberOfDownloadedCases,
+                }
```
Comment (Mathieu-Deharbe) [inline src/components/utils/caseUtils.ts:52]: A bit strange to format those 2 lines differently.

File: src/components/utils/caseUtils.ts [hunk 2]
```diff
@@ -17,3 +27,136 @@ export const downloadCases = async (uuids: string[]) => {
...
+        return (
+            messageFirstLine + '\n' + capitalizeFirstLetter(messageSecondLine)
+        );
+    };
+
+    const handleDownloadCases = async (selectedElements: any[]) => {
+        const casesUuids = selectedElements
+            .filter((element) => element.type === ElementType.CASE)
+            .map((element) => element.elementUuid);
+        await downloadCases(casesUuids);
```
Comment (Mathieu-Deharbe) [inline src/components/utils/caseUtils.ts:150]: I personnaly have nothing against async functions but Joris told me that Jon discouraged him from using it. Is there an official constraint or not?

File: src/components/utils/caseUtils.ts [hunk 3]
```diff
@@ -17,3 +27,136 @@ export const downloadCases = async (uuids: string[]) => {
...
+        if (casesUuids.length !== selectedElements.length) {
+            snackInfo({
+                messageTxt: buildPartialDownloadMessage(
+                    casesUuids.length,
+                    selectedElements
+                ),
+            });
+        }
```
Comment (Mathieu-Deharbe) [inline src/components/utils/caseUtils.ts:158]: So the snackbar is only displayed when there were non situations selected ? Too bad : it looks much nicer than the navigator notification.

File: src/translations/en.json [hunk 4]
```diff
@@ -257,8 +257,17 @@
   "country": "Country",
   "voltageRegulatorOn": "Voltage regulation",
   "PlannedActivePowerSetPoint": "Planning active power set point",
-  "downloadCases": "Download Case(s)",
-
+  "download.button": "Download",
```
Comment (Mathieu-Deharbe) [inline src/translations/en.json:260]: Why did you add a subcategory "button" here ? Seems to me that such a generic word could be reused anywhere.

Same remark for "download.message.filters": "filters",
  "download.message.lists": "lists",
  "download.message.studies": "studies",

**Review by Mathieu-Deharbe — APPROVED** [review]

File: src/components/utils/caseUtils.ts (hunk 2 above)
Comment (EstherDarkish) [reply src/components/utils/caseUtils.ts:150]: I reused code from @ne0ds and checked beforehand with him.
Maybe we should do some refactoring, but not in this pull request.

File: src/translations/en.json (hunk 4 above)
Comment (EstherDarkish) [reply src/translations/en.json:260]: Yes, there should be work done with the application's translation keys, they are over the place. Because I work on another ticket modifying them, I'd like to structure them better, and we have to start somewhere (so, let's start here).

File: src/components/utils/caseUtils.ts (hunk 3 above)
Comment (EstherDarkish) [reply src/components/utils/caseUtils.ts:158]: Yes

File: src/components/utils/caseUtils.ts (hunk 1 above)
Comment (EstherDarkish) [reply src/components/utils/caseUtils.ts:52]: Yes, ideally we should spell out numbers under 10, but that would be too much code for this anectdotical message

File: src/components/utils/caseUtils.ts [hunk 5]
```diff
@@ -1,4 +1,14 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
 import { downloadCase, getCaseOriginalName } from '../../utils/rest-api';
+import { ElementType } from '../../utils/elementType';
+import { useIntl } from 'react-intl';
+import { useSnackMessage } from '@gridsuite/commons-ui';
 
 export const downloadCases = async (uuids: string[]) => {
```
Comment (sBouzols) [inline src/components/utils/caseUtils.ts]: If it's used only locally now (because you export useDownloadUtils now) maybe remove this export here

Comment (EstherDarkish) [reply src/components/utils/caseUtils.ts]: Yes, done

**Review by sBouzols — APPROVED** [review]
We could do better for plural management and multiple values management for translation but ok for now before defining Translation guidelines.
Code Review OK
